dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreDbClient.cs

```diff
@@ -0,0 +1,260 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.VectorData;
+using Npgsql;
+using Pgvector;
+
+namespace Microsoft.SemanticKernel.Connectors.Postgres;
+
+/// <summary>
+/// An implementation of a client for Postgres. This class is used to managing postgres database operations.
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the <see cref="PostgresDbClient"/> class.
+/// </remarks>
+/// <param name="dataSource">Postgres data source.</param>
+/// <param name="schema">Schema of collection tables.</param>
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities", Justification = "We need to build the full table name using schema and collection, it does not support parameterized passing.")]
+public class PostgresVectorStoreDbClient(NpgsqlDataSource dataSource, string schema = PostgresConstants.DefaultSchema) : IPostgresVectorStoreDbClient
+{
+    private readonly NpgsqlDataSource _dataSource = dataSource;
+    private readonly string _schema = schema;
+
+    private IPostgresVectorStoreCollectionSqlBuilder _sqlBuilder = new PostgresVectorStoreCollectionSqlBuilder();
+
+    /// <inheritdoc />
+    public async Task<bool> DoesTableExistsAsync(string tableName, CancellationToken cancellationToken = default)
+    {
+        NpgsqlConnection connection = await this._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Can you have multiple connections open at the same time?

Reply:
Yes, NpgsqlDataSource is a connection pool, so when you ask it to OpenConnectionAsync, it pulls a free connection from the pool or creates one up to the pool max connections.